src/app/shared/dialogs/dialogs-resources-viewer.component.ts

```diff
@@ -1,11 +1,29 @@
+import { Router, ActivatedRoute } from '@angular/router';
 import { Component, Inject } from '@angular/core';
-import { MAT_DIALOG_DATA } from '@angular/material';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
+
 
 @Component({
-  template: `<planet-resources-viewer [isDialog]="true" [resourceId]="data.resourceId"></planet-resources-viewer>`
+  template: `
+  <div mat-dialog-content>
+    <planet-resources-viewer [isDialog]="true" [resourceId]="data.resourceId"></planet-resources-viewer>
+  </div>
+  <div mat-dialog-actions style ="float:right">
+      <button mat-raised-button  color="primary"   (click) = "viewResources()">View Resource</button>
+  </div>
+    `
 })
 export class DialogsResourcesViewerComponent {
 
-  constructor(@Inject(MAT_DIALOG_DATA) public data: any) {}
+  constructor(@Inject(MAT_DIALOG_DATA) public data: any, public dialogRef: MatDialogRef<DialogsResourcesViewerComponent>,
```

Comment:
The preferred format for a constructor with many injected services is:
```
constructor(
  private service1,
  private service2,
  private service 3
) {}
```

Please change so this matches the above format.

In addition make the `dialogRef` private.